pkg/cloudprovider/aws/aws_test.go

```diff
@@ -322,3 +323,51 @@ func TestAddASGTags_WithErrorResponse(t *testing.T) {
 	)
 	addASGTags(&mockNodeGroupConfig, &mockASG, awsCloudProvider)
 }
+
+func TestTerminateInstances_Success(t *testing.T) {
+	setupAWSMocks()
+
+	// Mock service call and error
+	awsCloudProvider, _ := newMockCloudProviderUsingInjection(
+		nil,
+		&test.MockAutoscalingService{},
+		&test.MockEc2Service{
+			TerminateInstancesOutput: &ec2.TerminateInstancesOutput{},
+			TerminateInstancesErr:    nil,
+		},
+	)
+
+	instance1 := "i-123456"
+	instances := []*string{&instance1}
+	mockNodeGroup.provider = awsCloudProvider
+
+	terminateOrphanedInstances(&mockNodeGroup, instances)
```

Comment:
Would you be able to explain what is actually tested here?

From what I can tell `terminateOrphanedInstances` is only going to fail if it panics, where these tests never set `terminateInstancesTries == maxTerminateInstancesTries` so it will always pass

The cloud provider API call only logs a warning, so I don't immediately see how this it's caught?
```
_, err := n.provider.ec2Service.TerminateInstances(&ec2.TerminateInstancesInput{
			InstanceIds: awsapi.StringSlice(instanceIds),
		})
		if err != nil {
			log.Warnf("failed to terminate instances %v", err)
		}
```

Lemme know if I missed something :)

Reply:
Yeah since this function doesn't return anything I just added a few test cases to get coverage for panics. I was originally going to have a case where `terminateOrphanedInstances` was called a few times in a row to trigger the fatal, but from [this](https://stackoverflow.com/questions/30688554/how-to-test-go-function-containing-log-fatal) stackoverflow thread it seemed like there was some debate over the value of testing for fatals